shared-libs/user-management/test/unit/users.spec.js

```diff
@@ -3020,14 +3149,27 @@ describe('Users service', () => {
       sinon.stub(roles, 'isOffline').returns(false);
 
       const updates = { token_login: true, phone: '+40 755 89-89-89' };
-      db.medic.get.resolves({
+      db.medic.get.onFirstCall().resolves({
```

Comment:
This particular test had some very sketchy things happening behind the scenes due to some pass-by-reference effects.  Basically, the asserts below were "working" because the arg objects were getting passed around and continued to be updated as the logic flow continued (because of the `Object.assign` calls in `getUpdatedUserDoc` and `getUpdatedSettingsDoc`).  When I replaced the `Object.assign` calls by spreading the data into a new object, that broke this test.  

However, I think the updated test code is more explicit and accurate anyway.